doc/source/developer_reference/output_processing_api.rst

```diff
@@ -3,14 +3,61 @@
-:ref:`Output Directory Structure<output_directory>` which can be performed by using WA's
-``RunOutput`` object.
-
-Example:
-
-If we have an existing WA output called ``wa_output`` in the current working
-directory we can initialize a ``RunOutput`` as follows:
+WA3 now has an output API that can be used to post process a runs
+:ref:`Output Directory Structure <output_directory>` which can be performed by using WA's
+:class:`RunOutput` object.
+
+This object provides a simple way to access the various aspects of the run. A
+few of the most useful attributes of the ``RunOutput`` object are as follows:
+
+- :confval:`artifacts` - A list containing any run level artifacts (for example
+  logs, output generated by output processors etc.).
+- :confval:`augmentations` - A list of the augmentations that were enabled
+  during the run.
+- :confval:`basepath` - The name of the base WA output folder.
+- :confval:`classifiers` - A dict mapping of any run level
+  :ref:`classifiers  <classifiers>`.
+- :confval:`configfile` - The relative file path to the config file generated
+  for the run.
+- :confval:`event_summary` - A condensed summary of any events that occurred
+  during the run.
+- :confval:`events` - A list of any events that occurred during the run like job
+  failures etc.
+- :confval:`failed_dir` - The relative file path to the directory containing any
+  failed jobs during the run.
+- :confval:`info` - A :class:`RunInfo` object that contains information about
+  the run itself for example it's duration, name, uuid etc.
+- :confval:`infofile` - The file location of the json serialized file of the
+  :class:`RunInfo` object.
+- :confval:`jobs` - A list of all the job outputs that were generated for the
+  run. These can be used to used to access more detailed information about
+  individual jobs for example their ``status``, any ``metrics`` or ``artifacts``
+  that were generated etc.
+- :confval:`list_workloads()` - A method that generated a list of the workload
+  names that were executed during the run.
+- :confval:`logfile` - A relative file path to the run log.
+- :confval:`metadir` - A relative file path to the metadata directory.
+- :confval:`metrics` - A list of :ref:`Metrics <metrics>` that were generated
+  for the overall run.
+- :confval:`run_config` - A :class:`RunConfiguration` object that can be used to
+  access all the configuration of the run itself, for example the
+  ``reboot_policy``, ``execution_order``, ``device_config`` etc.
+- :confval:`settings` - The :ref:`Meta Configuration <meta-configuration>` used
+  during the run.
+- :confval:`status` - The overall status of the run.
+- :confval:`target_info` - A :class:`TargetInfo` object which can be used to
+  access various information about the target that was used during the run for
+  example it's ``abi``, ``hostname``, ``os`` etc.
+- :confval:`targetfile` - The file location of the json serialized file of the
```

Comment:
I would re-arrange these fields (and in the other APIs as well) in order of relevance - `jobs` should be the first thing listed, followed by the various config objects. The various files and directory paths should be listed last, as we would not expect for them, apart from `basepath`, to be accessed directly (as their contents would have already been parsed into the other objects). I'm kind of ambivalent on whether they should be included at all here (i.e. whether we want to make them part of the API).